pkg/handlers/embedded_cluster_node_join_command_test.go

```diff
@@ -0,0 +1,279 @@
+package handlers
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+
+	gomock "github.com/golang/mock/gomock"
+	embeddedclusterv1beta1 "github.com/replicatedhq/embedded-cluster/kinds/apis/v1beta1"
+	"github.com/replicatedhq/kots/pkg/handlers/kubeclient"
+	"github.com/replicatedhq/kots/pkg/store"
+	mockstore "github.com/replicatedhq/kots/pkg/store/mock"
+	"github.com/replicatedhq/kots/pkg/util"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	bootstrapapi "k8s.io/cluster-bootstrap/token/api"
+	bootstraputil "k8s.io/cluster-bootstrap/token/util"
+	kbclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+type testNodeJoinCommandHarness struct {
+	name              string
+	kbClient          kbclient.Client
+	httpStatus        int
+	token             string
+	getRoles          func(t *testing.T, token string) ([]string, error)
+	embeddedClusterID string
+	validateBody      func(t *testing.T, h *testNodeJoinCommandHarness, r *GetEmbeddedClusterNodeJoinCommandResponse)
+}
+
+func TestGetEmbeddedClusterNodeJoinCommand(t *testing.T) {
+	scheme := runtime.NewScheme()
+	corev1.AddToScheme(scheme)
+	embeddedclusterv1beta1.AddToScheme(scheme)
+
+	tests := []testNodeJoinCommandHarness{
+		{
+			name:              "not an embedded cluster",
+			httpStatus:        http.StatusBadRequest,
+			embeddedClusterID: "",
+		},
+		{
+			name:              "store returns error",
+			httpStatus:        http.StatusInternalServerError,
+			embeddedClusterID: "cluster-id",
+			getRoles: func(*testing.T, string) ([]string, error) {
+				return nil, fmt.Errorf("some error")
+			},
+		},
+		{
+			name:              "store gets passed the provided token",
+			httpStatus:        http.StatusInternalServerError,
+			embeddedClusterID: "cluster-id",
+			token:             "some-token",
+			getRoles: func(t *testing.T, token string) ([]string, error) {
+				require.Equal(t, "some-token", token)
+				return nil, fmt.Errorf("some error")
+			},
+		},
+		{
+			name:              "bootstrap token secret creation succeeds and it matches returned K0SToken",
```

Comment:
Went a bit overboard with testing other (unrelated) scenarios, but thought would be useful + it was a good opportunity for me to see how the join process works. Happy to move these to a separate PR if folks prefer.